code/modules/reagents/chemistry/reagents/water.dm

```diff
@@ -365,17 +365,18 @@
 			M.SetStuttering(0)
 			M.SetConfused(0)
 			return
-	if(ishuman(M) && M.mind && M.mind.vampire && !M.mind.vampire.get_ability(/datum/vampire_passive/full) && prob(80))
+	var/datum/antagonist/vampire/vampire = M.mind.has_antag_datum(/datum/antagonist/vampire)
+	if(ishuman(M) && is_vampire(M) && vampire.get_ability(/datum/vampire_passive/full) && prob(80))
```

Comment:
This will runtime on null-mind humans, since you do a `mind` access on line 368.